docs/uenv-compilation-spack.md

```diff
@@ -0,0 +1,163 @@
+# Using uenvs as upstream Spack instances
+
+User-environments (uenvs) are built with [Spack] using the [Stackinator] tool. Therefore, an `uenv` is tightly copuled with [Spack] and can be used as an upstream [Spack] instance (see [Chaining Spack Installations] for more details).
+
+!!! note
+    While this guide tries to explain everything step-by-step, it might be difficult to follow without any knowledge of [Spack]. Please have a look at [Spack Basic Usage] for a short introduction to [Spack].
+
+The uenv of a supported application contains all the dependencies to build the software with a particular configuration. In [Spack], such configuration is defined by a [spec] (see [Spack Basic Usage] for more details). Most uenvs already provide modules or [Spack Filesystem Views] which allow to manually build the same configuration. However, it is possible that you want to build or develop an application with a different configuration. To avoid re-building the whole uenv, you can re-use what is already there and build your new configuration using [Spack]. 
+
+This guide explains a _developer workflow_ allowing to either build your own package with [Spack], or use [Spack] to build all the dependencies and provide a build environment for the package.
+
+!!! note
+    This guide assumes that you have a local installation of [Spack]. If you don't have [Spack] installed, follow [Spack Getting Started].
+
+!!! warning
+    Avoid installing [Spack] on `HOME`. Packages are installaed within the `spack/` folder, and you might quickly run out of space.
```

Comment:
The eternal question: do we already know where spack should be installed by users, instead of saying "don't use home"? This is not a blocking comment. We might have to live with this until we know better.

Reply:
@simonpintarelli suggested `$SCRATCH` in #92. Should we go for that for the time being, and then update the documentation when there is a final decision?